Add support for `uefi_data` in builders

Marcelotsvaz opened the pull request:
Add support for `uefi_data` to register an AMI with a pre-populated UEFI variable store.
This is necessary for Secure Boot and to use custom UEFI boot entries.
Such variable stores can be created with https://github.com/awslabs/python-uefivars, by using the `aws` format the base64 encoded variable store can be passed to `uefi_data`.

Available for chroot and ebssurrogate builders which use `RegisterImage` and support `boot_mode`.

Closes #253

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/packer-plugin-amazon?pullRequest=272)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/packer-plugin-amazon?pullRequest=272)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/packer-plugin-amazon?pullRequest=272).</sub>

lbajolet-hashicorp commented on builder/ebssurrogate/builder_test.go:
```diff
@@ -57,6 +57,67 @@ func TestBuilderPrepare_InvalidKey(t *testing.T) {
+	config["ami_name"] = "name"
+
+	// Test bad
+	config["uefi_data"] = "foo"
```
As with the chroot tests, I think this is a good candidate for subtests with a common function for executing them.

lbajolet-hashicorp reviewed (approved):
Hey @Marcelotsvaz,

Thanks for the PR, apologies we took this long to review this code.

I left a few comments on the tests, the code itself looks good to me.

It seems that because of the changes that have been merged since, this PR will need to be rebased on top of main before we can merge it.

Please let me know if you want me to take a jab at this, happy to help.

Thanks again!

Marcelotsvaz commented:
Rebased and refactored the tests into subtests.

I'm not very familiar with Go but I think that by subtests you meant like `TestBuilderPrepare_IMDSSupportValue` right?

I think `Fatalf` should be fine now that we're not checking multiple conditions in the same test.

lbajolet-hashicorp commented:
Hi @Marcelotsvaz,

Bang on regarding the subtests, the refactored ones look good to me, we can merge this code. Thanks for being so responsive, much appreciated.

micchickenburger commented:
Would it make sense to add this for the EBS builder as well?

Marcelotsvaz commented:
The [CreateImage](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateImage.html#API_CreateImage_RequestParameters) API call used by the EBS builder doesn't support the `UefiData` parameter as opposed to the [RegisterImage](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RegisterImage.html#API_RegisterImage_RequestParameters) API call used in the EBS surrogate and chroot builders. AFAIK `CreateImage` copies the UEFI variable store of the running instance into the new AMI, so you can just configure the variables manually with a Packer provisioner.
